### PR TITLE
Cachingdef

### PR DESCRIPTION
**Pull request by hypothesist**

### CODE REVIEW

**Comment by sils1297** [comment]
ack https://github.com/coala-analyzer/coala/pull/2539/commits/357d56ddc6edb1853489258dc5ddbe14ced3e6df https://github.com/coala-analyzer/coala/pull/2539/commits/011814e001f5e0a991f25ca876b021e59a54bee2

File: coalib/misc/CachingUtilities.py [hunk 1]
```diff
@@ -136,7 +137,13 @@ def get_settings_hash(sections):
     """
     settings = []
     for section in sections:
+        orig = copy.deepcopy(sections[section])
```
Comment (sils1297) [inline coalib/misc/CachingUtilities.py]: actually orig is the copy and you change then the original, rather do it the other way round, make a copy and change that one ;)

Comment (sils1297) [inline coalib/misc/CachingUtilities.py]: actually orig is the copy and you change then the original, rather do it the other way round, make a copy and change that one ;)

**Comment by sils1297** [comment]
unack https://github.com/coala-analyzer/coala/pull/2539/commits/9cd72171092ceaf0d9356cd93e8d86d07f885c6e

File: coalib/misc/CachingUtilities.py [hunk 2]
```diff
@@ -136,7 +137,13 @@ def get_settings_hash(sections):
...
+        if "disable_caching" in sections[section]:
```
Comment (sils1297) [inline coalib/misc/CachingUtilities.py]: you should give this a parameter `ignored_settings: list` or so

Comment (sils1297) [inline coalib/misc/CachingUtilities.py]: because this function isn't caching specific right?

File: tests/misc/CachingTest.py [hunk 3]
```diff
@@ -105,8 +105,8 @@ def test_caching_results(self):
                     coala.main,
                     "coala",
                     "-c", os.devnull,
+                    "--disable-caching",
```
Comment (sils1297) [inline tests/misc/CachingTest.py]: ahm this shouldn't be here, rght?

File: tests/processes/ProcessingTest.py [hunk 4]
```diff
@@ -91,7 +91,7 @@ def setUp(self):
 
     def test_run(self):
         self.sections['default'].append(Setting('jobs', "1"))
-        self.sections['default'].append(Setting('changed_files', True))
+        self.sections['default'].append(Setting('disable_caching', False))
```
Comment (sils1297) [inline tests/processes/ProcessingTest.py]: this defaults to false, just remove that line

File: tests/misc/CachingTest.py (hunk 3 above)
Comment (hypothesist) [inline tests/misc/CachingTest.py]: Coverage. Nobody should ever need `disable-caching`, but I'm adding it just in case there's an issue with caching

**Comment by sils1297** [comment]
ack https://github.com/coala-analyzer/coala/pull/2539/commits/c296cdde245f059f6a0e8d5b1f3524a48cfbe784

**Comment by hypothesist** [comment]
reack https://github.com/coala-analyzer/coala/pull/2539/commits/7d57784e0fd0f09ca36815b53498475713129420 https://github.com/coala-analyzer/coala/pull/2539/commits/01890c2b6288a8ed78c7d5a167a796164884b38b https://github.com/coala-analyzer/coala/pull/2539/commits/173281382071c4fb660457c100aa03c52b6fbc52

**Comment by sils1297** [comment]
@rultor merge

**Pull request merged**